src/execution.jl

```diff
@@ -118,6 +118,39 @@ for x in [:ExecutionEngine, :Interpreter, :JIT]
     end
 end
 
+Base.push!(engine::ExecutionEngine, mod::Module) = API.LLVMAddModule(engine.ref, mod.ref)
```

Comment:
Would it be useful to have `do` block syntax here, or do you usually push multiple modules at once?

Reply:
Could you elaborate?

Reply:
```
push!(engine, mod) do
    # do something with engine
    # implicit delete!(engine, mod)
end
```